src/testRunner/unittests/tsbuildWatch/publicApiImpliedNodeFormat.ts

```diff
@@ -0,0 +1,254 @@
+                    target: "es2020",
+                },
+            })
+        };
+        const sharedConfig: File = {
+            path: `${projectRoot}/shared/tsconfig.json`,
+            content: JSON.stringify({
+                extends:"./tsconfig.options.json",
+                compilerOptions: {
+                    composite: true
+                },
+            })
+        };
+        const sharedPackageJson: File = {
+            path: `${projectRoot}/shared/package.json`,
+            content: JSON.stringify({
+                name:"shared",
+                version:"1.0.0",
+                type:"commonjs",
+            })
+        };
+        const sharedIndex: File = {
+            path: `${projectRoot}/shared/index.ts`,
+            content: `export function f1() { }
+export class c { }
+export enum e { }
+// leading
+export function f2() { } // trailing`
+        };
+        const webpackConfig: File = {
+            path: `${projectRoot}/webpack/tsconfig.json`,
+            content: JSON.stringify({
+                compilerOptions: { composite: true, },
+                references: [{ path: "../shared/tsconfig.json" }]
+            })
+        };
+        const webpackIndex: File = {
+            path: `${projectRoot}/webpack/index.ts`,
+            content: `export function f2() { }
+export class c2 { }
+export enum e2 { }
+// leading
+export function f22() { } // trailing`
+        };
+        const commandLineArgs = ["--b", "--w", /*"--verbose"*/];
+        const { sys, baseline, oldSnap, cb, getPrograms } = createBaseline(createWatchedSystem([libFile, { ...libFile, path: "/a/lib/lib.es2020.d.ts" }, solution, sharedConfig, sharedConfigOptions, sharedPackageJson, sharedIndex, webpackConfig, webpackIndex], { currentDirectory: projectRoot }));
+
+        const writeToConsole = false; // MUST BE false WHEN NOT TESTING SINGULARLY
+        if (writeToConsole){
+            const origSysWrite = sys.write.bind(sys);
+            sys.write = (message: string)=>{
+                console.log(message);
+                origSysWrite(message);
+            };
+        }
+        type ImpliedNodeFormat = ModuleKind.CommonJS | ModuleKind.ESNext | undefined;
+        const impliedNodeFormatToString = (f: ImpliedNodeFormat)=>(f===undefined)? "undefined" : ModuleKind[f];
+        const lastImpliedNodeFormats = new Map<string,{impliedNodeFormat: ImpliedNodeFormat /*, gotImpledNodeFormat: ImpliedNodeFormat*/}>();
+        function printLastImpliedNodeFormats(){
+            lastImpliedNodeFormats.forEach((x,fileName)=>{
+                sys.write(`fileName:${fileName},impliedNodeFormat:${impliedNodeFormatToString(x.impliedNodeFormat)}`+sys.newLine);
+            });
+        }
+        // function printModuleResolutionCache(buildr: SolutionBuilder<EmitAndSemanticDiagnosticsBuilderProgram>){
+        //     const x = { moduleResolutionCache: buildr.getModuleResolutionCache() };

```

Comment:
what is this test testing? I expected to see getModuleResolutionCache somewhere, but it's commented out here

Reply:
This test is writing and changing package.json "type" fields (values "commonjs" or "module"), and checking that the change detected is propagated up to the `SourceFile["impliedNodeFormat"]` field - in the context of a SolutionBuilder watch build.  

The `getModuleResolutionCache` is apparently not used at all (and will not be used).  I didn't know that, and put it there as part of a question I asked  @sheetalkamat.  I will remove it.

Reply:
The commented out code referring to `getModuleResolutionCache` is removed in the now uploaded code.